### PR TITLE
Cache

### PR DESCRIPTION
**Pull request by Revenaunt**
Added a two-level caching system, first the memory cache is checked. On hit, the image is pulled from memory, on miss, the disk cache is checked. On hit the image is read from disk, on miss, the image is downloaded. DiskLruCache.java can be ignored, it is a library.

@bradmontgomery check it out when possible and let me know.

### CODE REVIEW

File: src/main/java/org/tndata/android/compass/model/Category.java [hunk 1]
```diff
@@ -185,14 +186,15 @@ public int compareTo(Category another) {
     public void loadIconIntoView(Context context, ImageView imageView) {
         String iconUrl = getIconUrl();
         if(iconUrl != null && !iconUrl.isEmpty()) {
-            ImageCache.instance(context).loadBitmap(imageView, iconUrl, false);
+            //ImageCache.instance(context).loadBitmap(imageView, iconUrl, false);
```
Comment (bradmontgomery) [inline src/main/java/org/tndata/android/compass/model/Category.java]: Did you mean to replace this with the new `ImageLoader`?

**Comment by bradmontgomery** [comment]
@Revenaunt If the app is closed, and then opened back up, does the cache get reset, or does the cache store images indefinitely?

File: src/main/java/org/tndata/android/compass/model/Category.java (hunk 1 above)
Comment (Revenaunt) [inline src/main/java/org/tndata/android/compass/model/Category.java]: I did, I'll go ahead and change it.

Comment (Revenaunt) [inline src/main/java/org/tndata/android/compass/model/Category.java]: @bradmontgomery I did, I'll go ahead and change it.

**Comment by Revenaunt** [comment]
@bradmontgomery The memory cache resets, the disk cache is persistent.

**Comment by bradmontgomery** [comment]
@Revenaunt cool. Once you include the update to `Category.java`, I'm giving this a :+1:. You can go ahead and merge the PR.

**Comment by Revenaunt** [comment]
@bradmontgomery fixed.

**Pull request merged**